src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/JsonFunctionCallUtilities.cs

```diff
@@ -14,40 +15,68 @@
+    bool HasDefaultValue,
+    object? DefaultValue,
+    bool IncludeSchemaUri,
+    bool DisallowAdditionalProperties,
+    bool IncludeTypeInEnumSchemas);
 
 namespace Microsoft.Extensions.AI;
 
 /// <summary>Provides a collection of static utility methods for marshalling JSON data in function calling.</summary>
-internal static partial class FunctionCallHelpers
+public static partial class JsonFunctionCallUtilities
 {
+    /// <summary>The uri used when populating the $schema keyword in inferred schemas.</summary>
+#pragma warning disable S1075 // URIs should not be hardcoded
+    private const string SchemaKeywordUri = "https://json-schema.org/draft/2020-12/schema";
+#pragma warning restore S1075 // URIs should not be hardcoded
+
     /// <summary>Soft limit for how many items should be stored in the dictionaries in <see cref="_schemaCaches"/>.</summary>
     private const int CacheSoftLimit = 4096;
 
     /// <summary>Caches of generated schemas for each <see cref="JsonSerializerOptions"/> that's employed.</summary>
     private static readonly ConditionalWeakTable<JsonSerializerOptions, ConcurrentDictionary<FunctionParameterKey, JsonElement>> _schemaCaches = new();
 
     /// <summary>Gets a JSON schema accepting all values.</summary>
-    private static JsonElement TrueJsonSchema { get; } = ParseJsonElement("true"u8);
+    private static readonly JsonElement _trueJsonSchema = ParseJsonElement("true"u8);
 
     /// <summary>Gets a JSON schema only accepting null values.</summary>
-    private static JsonElement NullJsonSchema { get; } = ParseJsonElement("""{"type":"null"}"""u8);
+    private static readonly JsonElement _nullJsonSchema = ParseJsonElement("""{"type":"null"}"""u8);
+
+    /// <summary>
+    /// Removes characters from a .NET member name that shouldn't be used in an AI function name.
+    /// </summary>
+    /// <param name="memberName">The .NET member name that should be sanitized.</param>
+    /// <returns>
+    /// Replaces non-alphanumeric characters in the identifier with the underscore character.
+    /// Primarily intended to remove characters produced by compiler-generated method name mangling.
+    /// </returns>
+    public static string SanitizeMemberName(string memberName) =>
+        InvalidNameCharsRegex().Replace(memberName, "_");
 
     /// <summary>Parses a JSON object into a dictionary of objects encoded as <see cref="JsonElement"/>.</summary>
     /// <param name="json">A JSON object containing the parameters.</param>
     /// <param name="parsingException">If the parsing fails, the resulting exception.</param>
     /// <returns>The parsed dictionary of objects encoded as <see cref="JsonElement"/>.</returns>
-    public static Dictionary<string, object?>? ParseFunctionCallArguments(string json, out Exception? parsingException)
+    public static Dictionary<string, object?>? ParseFunctionCallArguments([StringSyntax(StringSyntaxAttribute.Json)] string json, out Exception? parsingException)
```

Comment:
This is just following our usual practice of returning the most specific type. Any reason why we would want to abstract it away?

Reply:
> This is just following our usual practice of returning the most specific type. Any reason why we would want to abstract it away?

For future flexibility should we want to return a different kind of dictionary.

Reply:
Isn't the usual counterargument that we introduce an implicit dependency on the current implementation if it's public since users can do `result is Dictionary<string, object?>`?

Reply:
We say folks can't rely on concrete type beyond what's in the signature, and we have changed it. We do not consider it to be a breaking change.

Reply:
I don't see a strong reason why we'd want to move away from `Dictionary<string, object?>`. This is a deserialization routine that assigns ownership of the dictionary to the caller, so unless the signature of `FunctionCallContent` changes dramatically I don't think this would need to change either.

One approach I briefly experimented with was changing the return type to be `FunctionalCallContent` and have the method take additional functionName and callId parameters. While this did clean the API signature a bit, it made setting additional properties a bit more clunky at the call sites. It would serve to eliminate the concern you're raising here though. Thoughts?

Reply:
I think I like that better. Makes me wonder whether that helper should be on FunctionCallContent instead, but if we're trying to keep the JSON-ness separate, this makes sense.